plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -1,23 +1,148 @@
+import java.util.List;
+
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.ui.console.MessageConsoleStream;
+import org.eclipse.wst.server.core.IModule;
 import org.eclipse.wst.server.core.IServer;
+import org.eclipse.wst.server.core.model.IModuleResource;
+import org.eclipse.wst.server.core.model.IModuleResourceDelta;
 import org.eclipse.wst.server.core.model.ServerBehaviourDelegate;
 
+import com.google.cloud.tools.appengine.api.AppEngineException;
+import com.google.cloud.tools.appengine.api.devserver.AppEngineDevServer;
+import com.google.cloud.tools.appengine.api.devserver.DefaultRunConfiguration;
+import com.google.cloud.tools.appengine.api.devserver.DefaultStopConfiguration;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdkAppEngineDevServer;
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessExitListener;
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessStartListener;
+import com.google.cloud.tools.eclipse.appengine.localserver.Activator;
+
+/**
+ * A {@link ServerBehaviourDelegate} for App Engine Server executed via the Java App Management
+ * Client Library.
+ */
 public class LocalAppEngineServerBehaviour extends ServerBehaviourDelegate {
+  private LocalAppEngineStartListener localAppEngineStartListener;
+  private LocalAppEngineExitListener localAppEngineExitListener;
+  private AppEngineDevServer devServer;
+
+  public LocalAppEngineServerBehaviour () {
+    localAppEngineStartListener = new LocalAppEngineStartListener();
+    localAppEngineExitListener = new LocalAppEngineExitListener();
+  }
 
   @Override
   public void stop(boolean force) {
     setServerState(IServer.STATE_STOPPING);
-    // TODO stop server
+    // stop server
+    terminate();
     setServerState(IServer.STATE_STOPPED);    
   }
 
-  void setStarted() {
-    setServerState(IServer.STATE_STARTED);    
+  /**
+   * Convenience method allowing access to protected method in superclass.
+   */
+  @Override
+  protected IModuleResourceDelta[] getPublishedResourceDelta(IModule[] module) {
+    return super.getPublishedResourceDelta(module);
+  }
+
+  /**
+   * Convenience method allowing access to protected method in superclass.
+   */
+  @Override
+  protected IModuleResource[] getResources(IModule[] module) {
+    return super.getResources(module);
+  }
+
+  /**
+   * Returns runtime base directory. Uses temp directory.
+   */
+  public IPath getRuntimeBaseDirectory() {
+    return getTempDirectory(false);
   }
-  
-  void setStarting() {
-    setServerState(IServer.STATE_STARTING);    
+
+  /**
+   * @return the directory at which module will be published.
+   */
+  public IPath getModuleDeployDirectory(IModule module) {
+    return getRuntimeBaseDirectory().append(module.getName());
+  }
+
+  /**
+   * Convenience accessor to protected member in superclass.
+   */
+  public void setModulePublishState2(IModule[] module, int state) {
```

Comment:
The superclass, ServerBehaviourDelegate,  has a  final setModulePublishState method